server/controller/profController.js

```diff
@@ -10,52 +10,63 @@ var Users = require('../../app/collections/users');
 var Dogs = require('../../app/collections/dogs');
 var Walkers = require('../../app/collections/walkers');
 
-exports.displayProf = function(req, res) {
-  // console.log('req.body: ', req.body);
+module.exports = {
+  displayProf: function(req, res) {
+    console.log('getting profile data!');
+    console.log('req.params: ', req.params);
 
-  var username = req.params.username;
-  var profile = {}
-  // search the db for a particular username
-
-  console.log('getting profile data for ' + username + '!');
-  new User({username: username})
-  .fetch()
-  .then(function(user) {
-    if (!user) {
-      // if !found, do... something?
-      console.log('user not found!');
-      res.redirect('/'); // PLACEHOLDER
-      // res.send();  // should probably send something else
-    } else {
-      // if found: grab the user profile data
-      Object.assign(profile, user.toJSON());
-      var userId = user.get('id');
-
-      if (user.get('type') === 'Dog') {
-        console.log('grabbing dog profile: ', userId);
-        // console.log('from user model: ', user);
-        new Dog({userId: userId}).fetch()
-        .then(function(dog) {
-          Object.assign(profile, dog.toJSON());
-        })
-        .then(function() {
-          console.log('\nreturning: ', profile);
-          res.send(profile);
-        });
+    var username = req.params.username;
+    var profile = {}
+    // search the db for a particular username
+    new User({username: username})
+    .fetch()
+    .then(function(user) {
+      if (!user) {
+        // if !found, do... something?
+        console.log('user not found!');
+        res.redirect('/'); // PLACEHOLDER
+        // res.send();  // should probably send something else
       } else {
-        new Walker({userId: userId}).fetch()
-        .then(function(walker) {
-          Object.assign(profile, walker.toJSON());
-        })
-        .then(function() {
-          console.log('\nreturning: ', profile);
-          res.send(profile);
-        });
+        // if found: grab the user profile data
+        Object.assign(profile, user.toJSON());
+        var userId = user.get('userId');
+
+        if (user.get('isDog')) {
+          new Dog({userId: userId}).fetch()
+          .then(function(dog) {
+            Object.assign(profile, dog.toJSON());
+          });
+        } else {
+          new Walker({userId: userId}).fetch()
+          .then(function(walker) {
+            Object.assign(profile, walker.toJSON());
+          });
+        }
+        // build a bookshelf query and search the appropriate table
+        // res.redirect('/'); // PLACEHOLDER
+        res.send(profile);
```

Comment:
did you test this in postman or on the localhost? are you certain it will perform as intended even with async behavior?